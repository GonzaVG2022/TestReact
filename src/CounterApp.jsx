import  PropTypes  from "prop-types";
import { useState } from "react";

export const CounterApp = ({value}) => {
  
  const [counter, setCounter] = useState(value);
  const sumar = () => 
    setCounter( counter + 1 )
  
  const restar = () => 
    setCounter( counter - 1 )
  
  const reset = () => 
    setCounter( value )
  
  return (
    <>
    <h1>CounterApp</h1>
    {/* <h2> { value } </h2> */}
    <h1>{counter}</h1>

    <button 
      aria-label="sumar"
      onClick={sumar}
    >
      +1
    </button>
    <button 
      onClick={restar}
    >
      -1
    </button>
    <button 
      onClick={reset}
    >
      Reset
    </button>
    
    </>
  )
}
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
