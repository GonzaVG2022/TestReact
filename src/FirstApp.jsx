import  PropTypes  from "prop-types";



export const FirstApp = ({title, name,subTitle}) => {
  // console.log(props)
  
  return (
    <>
       
        <h1 data-testid='test-title'>{title}</h1>
        <h2>{name}</h2>
        <h3>{subTitle}</h3>
        <h3>{subTitle}</h3>
    
    </>
    
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired, //lo obligo a que lo envie
  name: PropTypes.string.isRequired
}//Propertis con types

// definir las prop por defecto
FirstApp.defaultProps = {
  title :'No hay titulo',
  subTitle: 'Holiiiis',
  name: 'No hay nombre'
}