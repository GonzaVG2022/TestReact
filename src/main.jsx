import React from 'react'
import  ReactDOM from 'react-dom/client'
import './styles.css'//styles globales
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp
            value= {10}
        />
       {/* <FirstApp
        title='Hola soy Flavio'
       /> */}

    </React.StrictMode>
)