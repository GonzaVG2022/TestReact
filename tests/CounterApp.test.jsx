import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Prueba en CounterApp', () => {
    const initialValue = 10
    test('debe hacer match con el snapshot', () =>{
        const {container} = render(
            <CounterApp
                value= {initialValue}
            />)
    expect(container).toMatchSnapshot()
            }
        );
    test('debe de retornar el valor de 100', () =>{
        render(<CounterApp value={100}/>);
        expect(screen.getByText(100)).toBeTruthy();

    })
    test('debe incrementar con el boton 1', () =>{
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('11')).toBeTruthy();

    })
    test('debe decrementar con el boton -1', () => {
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('9')).toBeTruthy();
    })
    test('debe retornar el reset', () => {
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('Reset'));
        screen.debug()//veo el estado del dom
        expect(screen.getByText('10')).toBeTruthy();
    })
    test('debe funcionar el boton Reset',() => {
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByRole('button',{name:'sumar'}))
        expect(screen.getByText('11')).toBeTruthy();
    })
});