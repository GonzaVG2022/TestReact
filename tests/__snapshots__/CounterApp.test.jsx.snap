// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en CounterApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h1>
    10
  </h1>
  <button
    aria-label="sumar"
  >
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
